oval-schemas/linux-definitions-schema.xsd

```diff
@@ -2718,6 +2718,110 @@
+                 <xsd:sequence>
+                   <xsd:element name="filter_key" type="oval-def:EntityObjectStringType" nillable="true">
+                     <xsd:annotation>
+                       <xsd:documentation>As described in the auditctl(8) manpage, the filter key is an arbitrary string of text that can be up to 31 bytes long. It can uniquely identify the audit records produced by a rule. You may have more than one key on a rule.</xsd:documentation>
+                       <xsd:documentation>If the xsi:nil attribute is set to true, all auditd rules must be present in the system characteristics (auditdline_item).</xsd:documentation>
+                     </xsd:annotation>
+                   </xsd:element>
+                   <xsd:element ref="oval-def:filter" minOccurs="0" maxOccurs="unbounded"/>
+                 </xsd:sequence>
+               </xsd:choice>
+             </xsd:sequence>
+           </xsd:extension>
+         </xsd:complexContent>
+       </xsd:complexType>
+     </xsd:element>
+     <xsd:element name="auditdline_state" substitutionGroup="oval-def:state">
+       <xsd:annotation>
+         <xsd:documentation>The auditdline_state element defines the different information that can be used to evaluate the auditd rules. This includes the filter key, the corresponding rule and the line number of the rule. Please refer to the individual elements in the schema for more details about what each represents.</xsd:documentation>
+       </xsd:annotation>
+       <xsd:complexType>
+         <xsd:complexContent>
+           <xsd:extension base="oval-def:StateType">
+             <xsd:sequence>
+               <xsd:element name="filter_key" type="oval-def:EntityStateStringType" minOccurs="0" maxOccurs="1">
+                 <xsd:annotation>
+                   <xsd:documentation>As described in the auditctl(8) manpage, the filter key is an arbitrary string of text that can be up to 31 bytes long. It can uniquely identify the audit records produced by a rule. You may have more than one key on a rule.</xsd:documentation>
+                 </xsd:annotation>
+               </xsd:element>
+               <xsd:element name="audit_line" type="oval-def:EntityStateStringType" minOccurs="0" maxOccurs="1">
+                 <xsd:annotation>
+                   <xsd:documentation>A rule written on a single line like returned by the auditctl -k command.</xsd:documentation>
+                 </xsd:annotation>
+               </xsd:element>
+               <xsd:element name="line_number" type="oval-def:EntityStateIntType" minOccurs="0" maxOccurs="1">
+                 <xsd:annotation>
+                   <xsd:documentation>The line number of the rule, which can be considered as the rule number regarding that there is one rule per line. This number starts at 1 which means that the number of the first rule returned is 1.</xsd:documentation>
+                 </xsd:annotation>
+               </xsd:element>
```

Comment:
The line number is a great element for `auditdline_state`, it can help check the order relationship between two audit rules. But checking the order relationship between 3 or 4 rules can be cumbersome. Is there a way we could facilitate checking the line order of the audit rules?

Sometimes it is interesting to check whether a particular audit rule is the last rule, for example `-e 2`.
Could we define a line number of `0`, or `-1` as the last audit rule?

Reply:
In the textfilecontent54_test, we introduced the concept of setting negative instance numbers.  -1 == last.  -2 == next-to-last, etc.

Reply:
Thank you for this feedback ! I think this is a good idea.  
Do you have any advice regarding setting negative instance numbers in the schema ? 
Is it just schema documentation or is there any constraint implementation needed ?

Reply:
It's just schema documentation.  From the textfilecontent54_object schema documentation for the instance entity:

```
The instance entity calls out a specific match of the pattern. It can have any integer value. If the
value is a non-negative integer, the index of the specific match of the pattern is counted from the
beginning of the set of matches of that pattern in the targeted file. The first match is given an
instance value of 1, the second match is given an instance value of 2, and so on. For non-negative
values, the 'less than' and 'less than or equal' operations imply the the object is operating only on
non-negative values. Frequently, this entity will be defined as 'greater than or equal' to 1 or 'greater
than' 0, either of which results in the object representing the set of all matches of the pattern.
Negative values are used to simplify collection of pattern match occurrences counting backwards
from the final match in the targeted file. To find the final match, use an instance of -1; the
penultimate match is found using an instance value of -2, and so on. For negative values, the
'greater than' and 'greater than or equal' operations imply the object is operating only on negative
values. For example, searching for instances greater than or equal to -2 would yield only the last
two maches.Note that the main purpose of the instance item entity is to provide uniqueness for
different textfilecontent_items that results from multiple matches of a given pattern against the
same file, and they will always have positive values.
```